File: euler_099.py
from time import time
from math import log10

with open("p099_base_exp.txt", "r") as f:
    raw = f.readlines()

powers = []

for i in range(len(raw)):
    p = raw[i].replace("\n", "")
    p = list(map(int, p.split(",")))
    powers.append(p)

start = time()

f = {n:powers[n][1] * log10(powers[n][0]) for n in range(len(powers))}

i = 0
m = f[i]

for j in range(len(powers)):
    if f[j] > m:
        i = j
        m = f[j]


print("Line with largest power: ", i + 1)

stop = time()
print("Time: " + str(stop - start))

File: euler_099.js
const start = Date.now();

import { readFileSync } from 'fs';


function syncReadFile(filename) {
    const contents = readFileSync(filename, 'utf-8');

    const arr = contents.split("\n");

    let pairs = [];
    for (const element of arr) {
        pairs.push(element.split(",").map(x => Number(x)));
    }

    return pairs;
}

const baseExponentPairs = syncReadFile('./euler_099.txt');

const deepCopy = [...baseExponentPairs];

deepCopy.sort((pair1, pair2) => Math.log(pair2[0]) * pair2[1] - Math.log(pair1[0]) * pair1[1]);

const end = Date.now();

console.log("Execute time: ", end - start, " Line number is: ", baseExponentPairs.indexOf(deepCopy[0]) + 1);